eq-author-api/schema/resolvers/utils.js

```diff
@@ -83,9 +83,21 @@ const getAnswerByValidationId = (ctx, validationId) =>
 const getAvailablePreviousAnswersForValidation = (ctx, validationId) => {
   const answer = getAnswerByValidationId(ctx, validationId);
   const currentPage = getPageByAnswerId(ctx, answer.id);
-  return getPreviousAnswersForPage(ctx.questionnaire, currentPage.id, false, [
-    answer.type,
-  ]);
+
+  const previousAnswers = getPreviousAnswersForPage(
+    ctx.questionnaire,
+    currentPage.id,
+    false,
+    [answer.type]
+  );
+
+  if (answer.type === UNIT) {
```

Comment:
Can we push this down into the `getPreviousAnswersForPage`?

Reply:
Not without refactoring the way it works, that function just gets all the previous answers for a given page rather than the previous answers for a given answer.